### PR TITLE
Bump Falco to 1.0.11 and add recording rules removed for K8s >=1.14.0

### PR DESCRIPTION
**Pull request by frankreno**
###### Description

This PR bumps falco to 1.0.11 (tested on non managed k8s 1.16, EKS 1.14).  It also adds a new template to add in recording rules that will only be rendered if the k8s version is >=1.14.0.  This is a short term fix to fix missing metrics that the K8s apps depend on while we plan the long term fix of supporting mixins.

###### Testing performed

- [ ] ci/build.sh
- [X] Redeploy fluentd and fluentd-events pods
- [X] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by samjsong** [comment]
I was under the impression that these additional recording rules were only required for EKS >= 1.14.0, is adding these rules safe for non-EKS users?

File: deploy/helm/sumologic/templates/prometheus-rules.yaml [hunk 1]
```diff
@@ -0,0 +1,189 @@
+{{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
+{{- if semverCompare ">=1.14.0-0" $kubeTargetVersion }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
```
Comment (rvmiller89) [inline deploy/helm/sumologic/templates/prometheus-rules.yaml]: are we not supporting yaml based install for these? (this template is Helm-only).

**Comment by frankreno** [comment]
@samjsong no it is a general k8s issue with 1.14, not specific to anything other than the version of k8s.  See #268

File: deploy/helm/sumologic/templates/prometheus-rules.yaml (hunk 1 above)
Comment (frankreno) [reply deploy/helm/sumologic/templates/prometheus-rules.yaml]: ahh good point, let me add that.

Comment (rvmiller89) [reply deploy/helm/sumologic/templates/prometheus-rules.yaml]: actually I might be confused, I see it's added as a section to the generated `fluentd-sumologic.yaml.tmpl`

**Comment by djsly** [comment]
LGTM, thanks for adding that as part of your official chart.  Once merged, I will be upgrading my colleciton solution

**Comment by frankreno** [comment]
@samjsong @rvmiller89 - found a better way to do this.  Single prop change in the values.yaml can force the pre 1.14 rules to render. Mind reviewing when you get a chance so we can get this merged?

File: deploy/kubernetes/fluentd-sumologic.yaml.tmpl [hunk 2]
```diff
@@ -656,3 +656,192 @@ spec:
           value: "100k"
         - name: TOTAL_LIMIT_SIZE
           value: "128m"
+---
+# Source: sumologic/templates/prometheus-rules.yaml
```
Comment (rvmiller89) [inline deploy/kubernetes/fluentd-sumologic.yaml.tmpl]: Huh, I think you'll need to open a new PR :( CI doesn't seem smart enough to remove this

Comment (frankreno) [reply deploy/kubernetes/fluentd-sumologic.yaml.tmpl]: Boo.  will do

**Comment by frankreno** [comment]
Opened #344

**Pull request closed without merging**